tests/sentry/api/helpers/test_group_index.py

```diff
@@ -1055,3 +1057,61 @@ def test_user_in_reassigned_team(self):
             user_id=user2.id,
             reason=GroupSubscriptionReason.assigned,
         ).exists()
+
+
+class DeleteGroupsTest(TestCase):
+    @patch("sentry.signals.issue_deleted.send_robust")
+    def test_delete_groups_simple(self, send_robust: Mock):
+        groups = [self.create_group(), self.create_group()]
+        group_ids = [group.id for group in groups]
+        request = self.make_request(user=self.user, method="GET")
+        request.user = self.user
+        request.GET = QueryDict(f"id={group_ids[0]}&id={group_ids[1]}")
+        hashes = ["0" * 32, "1" * 32]
+        for i, group in enumerate(groups):
+            GroupHash.objects.create(project=self.project, group=group, hash=hashes[i])
+            add_group_to_inbox(group, GroupInboxReason.NEW)
+
+        search_fn = Mock()
+        delete_groups(request, [self.project], self.organization.id, search_fn)
+
+        assert (
+            len(GroupHash.objects.filter(project_id=self.project.id, group_id__in=group_ids).all())
+            == 0
+        )
+        assert (
+            len(GroupInbox.objects.filter(project_id=self.project.id, group_id__in=group_ids).all())
+            == 0
+        )
+        assert send_robust.called
+
+    @with_feature("projects:similarity-embeddings-delete-by-hash")
+    @patch("sentry.tasks.delete_seer_grouping_records.logger")
+    @patch("sentry.signals.issue_deleted.send_robust")
+    def test_delete_groups_deletes_seer_records_by_hash(self, send_robust: Mock, mock_logger: Mock):
+        groups = [self.create_group(), self.create_group()]
+        group_ids = [group.id for group in groups]
+        request = self.make_request(user=self.user, method="GET")
+        request.user = self.user
+        request.GET = QueryDict(f"id={group_ids[0]}&id={group_ids[1]}")
+        hashes = ["0" * 32, "1" * 32]
+        for i, group in enumerate(groups):
+            GroupHash.objects.create(project=self.project, group=group, hash=hashes[i])
+            add_group_to_inbox(group, GroupInboxReason.NEW)
+
+        search_fn = Mock()
+        delete_groups(request, [self.project], self.organization.id, search_fn)
+
+        assert (
+            len(GroupHash.objects.filter(project_id=self.project.id, group_id__in=group_ids).all())
+            == 0
+        )
+        assert (
+            len(GroupInbox.objects.filter(project_id=self.project.id, group_id__in=group_ids).all())
+            == 0
+        )
+        assert send_robust.called
+        mock_logger.info.assert_called_with(
+            "calling seer record deletion by hash",
+            extra={"project_id": self.project.id, "hashes": hashes},
+        )
```

Comment:
In addition to asserting on the logger, it might be be worthwhile to assert on the task being called.

Reply:
thanks! I'll follow up with this after just because I want to get this deployed faster

Reply:
just followed up [here](https://github.com/getsentry/sentry/pull/73790)!